src/content/docs/reference/driver-guide.md

```diff
@@ -0,0 +1,452 @@
+---
+title: Membrane Driver Guide
+---
+
+## 1. Understanding the Driver's Role
+
+A driver serves as a bridge between Membrane and external APIs. We structure drivers to expose API functionality as nodes in your Membrane graph, making it easy for other programs to interact with external services.
+
+We have drivers for popular APIs like [GitHub](https://www.membrane.io/share/membrane/github) and [Slack](https://www.membrane.io/share/membrane/slack). Drivers are open source and anyone can create one. Once you get the hang of using Membrane drivers, you won’t have to worry much about the quirks of individual APIs—the standard driver structure and syntax will feel familiar.
+
+Drivers are regular Membrane programs so if you’ve written some code on Membrane you’re already half way there!
+
+For this guide, we'll walk through the [Resend  driver](https://www.membrane.io/share/membrane/resend) code.
+
+We'll demonstrate how to:
+
+- Create an interface for the API that uses Membrane state to store the API key
+- Structure your code using common Membrane patterns (Collections, Resources, grefs)
+- Follow consistent driver design principles and best practices
+
+Resend provides a modern email API for developers. For more details about their API, check out their [docs](https://resend.com/docs/introduction).
+
+## 2. Core Components
+
+<br/>
+
+### File Organization
+
+A typical Membrane driver should have this structure:
+
+```
+resend/
+├── index.ts        # Main driver code and exports
+├── helpers.ts      # API helper and utilities
+├── tests.ts        # Test cases
+├── memconfig.json  # Auto-generated schema
+└── README.md       # Documentation
+```
+
+### Configuration
+
+Every driver needs basic configuration and status checks. We add those in the `helpers.ts` file. Here's how we implement this for Resend:
+
+```js
+// helpers.ts
+import { state } from "membrane";
+
+// We use Membrane's state to persist the API key between program runs.
+export function status() {
+// The [text](:action) syntax makes the action invocable in the Membrane Navigator
```

Comment:
Not introduced in this PR, but looks like our code blocks aren't properly handling overflow. @juancampa I saw that you [added this](https://github.com/membrane-io/docs/blob/75fa65c3435be60effbb6d05fac3f4df589263af/src/styles/base.css#L166-L172), but it might be more of a footgun than it's worth?

<img width="640" alt="Screenshot 2024-12-09 at 5 02 23 PM" src="https://github.com/user-attachments/assets/a7646db8-169a-4de3-8288-f97225753ff4">